OpenMTRDemo/Filters/BaseFilter.cs

```diff
@@ -79,20 +80,11 @@ private void moveButton_Click(object sender, EventArgs e)
                 }
                 source.AddRange(reorder.ToArray());
             }
-            catch (NullReferenceException) { }
-            catch (IndexOutOfRangeException) { }
-        }
-
-        public abstract class BaseFilterListable
-        {
-            public string Name;
-            public ExpandedImageForm Form;
-            public FlowLayoutPanel Panel;
-            public abstract BaseFilter Instance();
-            public override string ToString()
+            catch (NullReferenceException)
             {
-                return Name;
+                MessageBox.Show("The filter was not declared properly for this button to work.");
             }
+            catch (IndexOutOfRangeException) { }
```

Comment:
Not sure of the likelihood of this one being thrown since we directly control the indices based on how many filters the user adds/takes away. If it is a possibility add an error message here as well. 

Also I just noticed this, when you open up the expanded image form the user is able to click the add button with a blank filter causing an error. Could we possibly use a "Selected Index changed" event handler to turn the add button on once a filter has been selected?

Reply:
This exception is thrown if the filter is already on the bottom and you try to move it down or already on the top and you try to move it up. I can add an error message, but I honestly think it would be intrusive.
Sure, I'll get on that add button thing.

Reply:
I've implemented a function that disables the each button when the button wouldn't work, rendering the point moot.